components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -844,6 +847,10 @@ func startAPIEndpoint(ctx context.Context, cfg *Config, wg *sync.WaitGroup, serv
 	defer wg.Done()
 	defer log.Debug("startAPIEndpoint shutdown")
 
+	if cfg.isHeadless() {
+		return
+	}
```

Comment:
headless tasks also need the API endpoint to satisfy the readiness probe, and because we listen to the headless tasks logs this way.